1/backend/src/resolvers.js

```diff
@@ -1,67 +1,268 @@
 const jwt = require('jsonwebtoken');
 const fs = require('fs');
+const {uuid} = require('uuidv4');
 
-data = require('./data');
+let data;
 
 let SECRET_KEY = fs.readFileSync('./src/key/secret.key', 'utf8');
 
 const resolvers = {
     Query: {
-        todos: () => JSON.parse(JSON.stringify(data.todos))
+        todos: async (parent, args, context) => {
+            // JSON.parse(JSON.stringify(data.todos))
+            if (args.limit != "" && args.limit != null) {
+                const {driver} = context
+                let getTodoCypher
+
+                getTodoCypher = `
+                    MATCH (todo:Todo)
+                    RETURN todo.id, todo.message, todo.completed LIMIT $limit
+                `
+
+
+                const session = driver.session()
+                try {
+                    data = await session.run(getTodoCypher, {limit: args.limit})
+                    const todos = await data.records.map(record => ({
+                        id: record.get('todo.id'),
+                        message: record.get('todo.message'),
+                        completed: record.get('todo.completed')
+
+                    }))
+                    json_data = JSON.parse(JSON.stringify(todos))
+                    return json_data
+                } finally {
+                    await session.close()
+                }
+            }
+        },
+        completedTodos: async (parent, args, context) => {
+            const {driver} = context
+            const getCompletedTodosCypher = `
+                    MATCH (todo:Todo) WHERE todo.completed = true 
+                    RETURN todo.id, todo.message, todo.completed ORDER BY todo.message
+             `
+            const session = driver.session()
+            try {
+                data = await session.run(getCompletedTodosCypher)
+                const todos = await data.records.map(record => ({
+                    id: record.get('todo.id'),
+                    message: record.get('todo.message'),
+                    completed: record.get('todo.completed')
+
+                }))
+                json_data = JSON.parse(JSON.stringify(todos))
+                return json_data
+            } finally {
+                await session.close()
+            }
+        }
     },
     Mutation: {
-        addTodo: (parent, args, context) => {
-            if(args.message != "" && args.message != null && context.token) {
-                let newTodo = {
-                    id: findNextId(),
+        addTodo: async (parent, args, context) => {
+            if (args.message != "" && args.message != null && context.token) {
+                const {driver} = context
+                const createTodoCypher = `
+                    CREATE (todo:Todo {params})
```

Comment:
```suggestion
                    CREATE (todo:Todo)
                    SET todo += $params
```
The other syntax seems to be deprecated.